frontend/src/Judge/components/VideoSubmission.js

```diff
@@ -1,5 +1,23 @@
 import React from 'react'
+import ReactPlayer from 'react-player'
+import styled from 'styled-components'
 
-const VideoSubmission = props => <div>Video Submission</div>
+// Video Providers
+const youtubeBaseURL = 'https://www.youtube.com/watch?v='
+const vimeoBaseURL = 'https://vimeo.com/'
+
+const videoStyle = {
+  minHeight: '500px',
+}
+
+const VideoSubmission = props => (
```

Comment:
Here's the component w/ the centering fix you asked me about yesterday – renders the player at the default size and prevents the player from having black bars.

```jsx
import React from 'react'
import PropTypes from 'prop-types'
import ReactPlayer from 'react-player'
import styled from 'styled-components'

// Video Providers
const YOUTUBE_BASE_URL = 'https://www.youtube.com/watch?v='
const VIMEO_BASE_URL = 'https://vimeo.com/'

const PlayerContainer = styled.div`
  margin-top: 50px;

  /* Center the top-most 'div' that ReactPlayer wraps around the 'iframe'  */
  & > div {
    margin: 0 auto;
  }
`

const VideoSubmission = props => (
  <PlayerContainer>
    {props.provider === 'youtube' ? (
      <ReactPlayer url={`${YOUTUBE_BASE_URL}${props.videoId}`} playing controls />
    ) : props.provider === 'vimeo' ? (
      <ReactPlayer url={`${VIMEO_BASE_URL}${props.videoId}`} playing controls />
    ) : null}
  </PlayerContainer>
)

VideoSubmission.propTypes = {
  provider: PropTypes.string.isRequired,
  videoId: PropTypes.string.isRequired
}

export default VideoSubmission

```

Reply:
Also, with your styles and my styles, when you vote on a video submission, the voting panel shifts down on the page when the alert appears. You should hunt down what's causing that and prevent the voting panel from moving.

Reply:
I do like how your styles make the video responsive. The 500px min-height, however, feels a bit tight on my laptop (full screen window), and not a fan of the black bars:

<img width="1192" alt="screen shot 2018-03-20 at 8 26 03 am" src="https://user-images.githubusercontent.com/13719429/37654481-958889fc-2c18-11e8-879f-11b1cd210759.png">